website/docs/r/automation_account.html.markdown

```diff
@@ -72,7 +72,7 @@ An `encryption` block supports the following:
 
 * `key_source` - (Optional) The source of the encryption key. Possible values are `Microsoft.Automation` and `Microsoft.Keyvault`.
 
-* `key_vault_key_id` - (Required) The ID of the Key Vault Key which should be used to Encrypt the data in this Automation Account.
+* `key_vault_key_id` - (Optional) The ID of the Key Vault Key which should be used to Encrypt the data in this Automation Account. Required when `key_source` is set to `Microsoft.Keyvault`.
```

Comment:
whilst it's unfortunate that we already have.. we shouldn't be exposing the `key_source` field here - we can infer that this should be set when `key_vault_key_id` is configured (meaning this block no longer needs to be marked as `computed`) - so could we update this to fix that?

Reply:
That's make sense! but i wonder what if other `key_source` type is supported in the future? and does `shouldn't be exposing the key_source field` means we should mark `key_source` as deprecated?

Reply:
yes, we should default it to Microsoft.Keyvault, and mark it as optional